### PR TITLE
Enable support for uint32 format to be consumed by SFPU (issue https://github.com/tenstorrent-metal/tt-metal/issues/4624) 

### PR DESCRIPTION
**Pull request by ttmtrajkovic**
Added support for UINT32 format.
Added fp32 dest view for unary and untilize op
added identity_uint32 test as an example for using uint32 format in sfpu

### CODE REVIEW

File: ttnn/ttnn/operations/core.py [hunk 1]
```diff
@@ -650,7 +650,7 @@ def to_layout(tensor, layout: ttnn.Layout, dtype: ttnn.DataType = None):
         raise RuntimeError(f"Unsupported layout conversion from {tensor.layout} to {layout}")
 
     is_on_device = ttnn.is_tensor_storage_on_device(tensor)
-    if is_on_device and tensor.dtype not in {ttnn.bfloat16, ttnn.bfloat8_b, ttnn.bfloat4_b}:
+    if is_on_device and tensor.dtype not in {ttnn.bfloat16, ttnn.bfloat8_b, ttnn.bfloat4_b, ttnn.float32, ttnn.uint32}:
```
Comment (cfjchu) [inline ttnn/ttnn/operations/core.py:653]: thanks for updating this... uint16 and int32 should also be added here too, right?

File: tests/ttnn/unit_tests/operations/test_unary.py [hunk 2]
```diff
@@ -27,6 +27,36 @@ def run_unary_test(device, h, w, ttnn_function, torch_function, pcc=0.9999):
...
+    output_tensor = ttnn.from_device(output_tensor)
+    output_tensor = ttnn.to_torch(output_tensor)
+
+    assert_with_pcc(torch_output_tensor, output_tensor, pcc)
+
+
+@pytest.mark.parametrize("h", [64])
+@pytest.mark.parametrize("w", [128])
+def test_fp32_uint32(device, h, w):
+    run_identity_test(device, h, w, ttnn.float32, pcc=0.9998)
```
Comment (cfjchu) [inline tests/ttnn/unit_tests/operations/test_unary.py:57]: Looks like you're not actually going to get any coverage on `ttnn.uint32` and `ttnn.uint16`. We should add a param sweep here on `data_type` so nobody breaks your changes.

File: tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp [hunk 3]
```diff
@@ -137,27 +139,28 @@ struct EltwiseUnary {
...
 operation::ProgramWithCallbacks eltwise_unary_single_core(
-    const Tensor& a, Tensor& output, const std::vector<UnaryWithParam> op_chain);
+    const Tensor& a, Tensor& output, const std::vector<UnaryWithParam> op_chain, bool fp32_dest_acc_en);
 
 inline Tensor run_eltwise_unary(
     const Tensor& input_tensor,
     std::vector<UnaryWithParam> ops_chain,
     const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG) {
     TT_FATAL(ops_chain.size() > 0, "At least 1 unary op must be specified");
+    bool fp32_dest_acc_en = input_tensor.get_dtype() == DataType::UINT32;       // MT: Currently only uint32 is moved to DST directly, fp32 is converted to fp16b
```
Comment (cfjchu) [inline tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp:153]: can float32 and int32 be made to do this unpack to dest directly as well, right? Is the problem just how to pipe that config down to the op?

File: ttnn/ttnn/operations/core.py (hunk 1 above)
Comment (ttmtrajkovic) [reply ttnn/ttnn/operations/core.py:653]: yes, I will update uint16, but int32 isn't exposed in the ttnn yet so I can't add it. 
I do plan to get to it so I will add it then

File: tests/ttnn/unit_tests/operations/test_unary.py (hunk 2 above)
Comment (ttmtrajkovic) [reply tests/ttnn/unit_tests/operations/test_unary.py:58]: This is stale code, I've updated it to cover only `ttnn.uint32`.
I can add a param sweep as well as you suggested.

File: tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp (hunk 3 above)
Comment (ttmtrajkovic) [reply tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp:153]: Yes, float32 and int32 should and will be part of this, but at this point float32 can't be preserved from unpacker's point of view do to some missing changes how in the `data_format.cpp`. Currently, fp32 as unpacker src format is recognized and unpacker dst format is being selected automatically as float16b, which is OK for Grayskull but not for wormhole. We need an additional parameter in the compile flow that will be enabled only for Wormhole and that will "tell" the compiler to preserve fp32. I will attend to this later as no one has requested it yet, uint32 is more important and it works.
Int32 should be part of this as well, but it's currently not supported as ttnn format.

Comment (cfjchu) [reply tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp:153]: Great, thanks! fyi @davorchap

**Review by cfjchu — APPROVED** [review]

**Review by rtawfik01 — APPROVED** [review]

**Pull request merged**